transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java

```diff
@@ -0,0 +1,338 @@
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TRANSFER_METHOD_CURRENCY;
+
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModel;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.hyperwallet.android.model.HyperwalletErrors;
+import com.hyperwallet.android.model.transfer.Transfer;
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.model.user.HyperwalletUser;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.transfer.repository.TransferRepository;
+import com.hyperwallet.android.ui.transfermethod.repository.TransferMethodRepository;
+import com.hyperwallet.android.ui.user.repository.UserRepository;
+
+import java.util.UUID;
+
+/**
+ * Create Transfer View Model
+ */
+public class CreateTransferViewModel extends ViewModel {
+
+    private static final String CLIENT_IDENTIFICATION_PREFIX = "HW-MBLA-";
+
+    private final TransferRepository mTransferRepository;
+    private final TransferMethodRepository mTransferMethodRepository;
+    private final UserRepository mUserRepository;
+
+    private final MutableLiveData<HyperwalletTransferMethod> mTransferDestination = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mTransferAvailableFunds = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mIsLoading = new MutableLiveData<>();
+    private final MutableLiveData<Boolean> mIsCreateQuoteLoading = new MutableLiveData<>();
+    private final MutableLiveData<Transfer> mQuoteAvailableFunds = new MutableLiveData<>();
+    private final MutableLiveData<Transfer> mQuoteTransfer = new MutableLiveData<>();
+    private final MutableLiveData<String> mTransferAmount = new MutableLiveData<>();
+    private final MutableLiveData<String> mTransferNotes = new MutableLiveData<>();
+
+    // initialisation/loading error
+    private Observer<Event<HyperwalletErrors>> mLoadErrorEventObserver;
+    private final MutableLiveData<Event<HyperwalletErrors>> mLoadErrorEvent = new MutableLiveData<>();
+    private final MutableLiveData<Event<HyperwalletErrors>> mLoadErrors = new MutableLiveData<>();
+
+    // create quote error
+    private Observer<Event<HyperwalletErrors>> mQuoteEventObserver;
+    private final MutableLiveData<Event<HyperwalletErrors>> mQuoteErrorEvent = new MutableLiveData<>();
+    private final MutableLiveData<Event<HyperwalletErrors>> mQuoteErrors = new MutableLiveData<>();
+
+    /**
+     * Initialize Create Transfer View Model with designated transfer source token
+     *
+     * @param sourceToken              Source token that represents either from Wallet, Bank or other EA tokens
```

Comment:
Please use user token or prepaid card token - avoid using terms like wallet/EA as they're used internally only. Also instead of HW use Hyperwallet